dial_test.go

```diff
@@ -16,7 +16,7 @@ import (
 	"github.com/saucelabs/forwarder/log/stdlog"
 )
 
-func TestResolverLookupHost(t *testing.T) {
```

Comment:
I'd like to see a test for context cancelation.

Reply:
Good idea.